extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,242 @@
+
+    const options = Array.from(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      handleError('Improper number of options');
+    }
+
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-prompt-container')
+      .appendChild(prompt);
+
+    options.forEach(this.configureOption_);
+
+    if (this.element.children.length !== 0) {
+      handleError('Too many children');
+    }
+  }
+
+  /**
+   * @private
+   * Creates an option container with option content,
+   * adds styling, answer choices, and percentage containers,
+   * and adds it to the shadow DOM.
+   *
+   * @param {HTMLOptionElement} option
+   */
+  configureOption_(option) {
+    // Transfer the option information into a span
+    const convertedOption = document.createElement('span');
+    convertedOption.textContent = option.textContent;
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    option.remove();
+
+    // Set up the class and add it to the shadow root
+    convertedOption.setAttribute('class', 'i-amp-story-quiz-option');
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-option-container')
+      .appendChild(convertedOption);
+
+    // Create a container for the answer choice and add it to the shadow root
+    const answerChoice = document.createElement('span');
+    answerChoice.textContent = this.answerChoiceOptions_.shift();
+    answerChoice.setAttribute('class', 'i-amp-story-quiz-answer-choice');
+    convertedOption.prepend(answerChoice);
+
+    // Create a container for the percentage and add it to the shadow root
+    const percentageBox = document.createElement('span');
+    percentageBox.setAttribute('class', 'i-amp-story-quiz-percentage');
+    convertedOption.append(percentageBox);
+  }
+
+  /**
+   * @private
+   * Temporary method to generate random percentages
+   *
+   * @return {Array<number>}
+   */
+  TEMPgenerateRandomPercentages_() {
+    const percentages = [];
+    let pool = 100;
+    let random;
+
+    const numOptions = Array.from(this.element.querySelectorAll('option'))
+      .length;
+    for (let i = 0; i < numOptions - 1; i++) {
+      random = Math.floor(Math.random() * pool);
+      percentages.push(random);
+      pool -= random;
+    }
+    percentages.push(pool);
+    return percentages;
+  }
+
+  /**
+   * @private
+   * Sets the percentage of an option and adjusts the background accordingly.
+   * @param {Node} option
+   * @param {number} percentage
+   */
+  setOptionPercentage_(option, percentage) {
+    let backgroundString;
+    if (
+      option.getAttribute('class').includes('i-amp-story-quiz-option-selected')
+    ) {
+      const colorPrefix = option.hasAttribute('correct') ? '' : 'in';
+      backgroundString = `linear-gradient(90deg, var(--${colorPrefix}correct-color-shaded) ${10 +
+        (9 * percentage) / 10}%, var(--${colorPrefix}correct-color) ${10 +
```

Comment:
Can you document what the `10+ (9 * percentage) / 10` pattern yields?  And maybe factor it out into a separate function?

Reply:
Done